File: LevelAndroidSDK/sdk/src/main/java/com/theshopatvsp/levelandroidsdk/ble/model/constants/BatteryState.java
package com.theshopatvsp.levelandroidsdk.ble.model.constants;

/**
 * Created by andrco on 6/12/16.
 */
public enum BatteryState {
    Charging(1), Discharging(2), Charged(4);

    BatteryState(int state) {
        this.state = state;
    }

    private int state;

    public int getState() {
        return state;
    }

    public static BatteryState getByState(int state) {
        for (BatteryState batteryState : values()) {
            if (batteryState.getState() == state) {
                return batteryState;
            }
        }

        return null;
    }
}

File: LevelIOSSdkExample/LevelIOSSdkExample/LevelIOSSdk/model/BatteryState.swift
//
//  BatteryState.swift
//  LevelIOSSDK
//
//  Created by Andrew Cook on 6/22/16.
//  Copyright © 2016 TheShop. All rights reserved.
//

import Foundation

enum BatteryState: Int {
    case Charging = 1
    case Discharging = 2
    case Charged = 4
    case BatteryError = 5
    
    var description: String {
        switch self {
        case .Charged:
            return "Charged"
        case .Charging:
            return "Charging"
        case .Discharging:
            return "Discharging"
        case .BatteryError:
            return "Battery Error"
        }
    }
}